%!TEX root = ../thesis.tex
\chapter[Diffusion MRI protocols for single fibre structures]{Optimized diffusion MRI protocols for estimating axon diameter with known fibre orientation}

In the previous chapters we have investigated two very different techniques in \gls{DTI} and \gls{QSI}. \gls{DTI} only requires relatively little amounts of data, but its underlying model is based on the assumption that all observed diffusion is Gaussian. Clearly, this is an over-simplification of the complex diffusion mechanisms present in biological tissue. \gls{QSI} on the other hand makes no assumption about the diffusion process and can theoretically reveal even the most complex diffusion displacement profiles. \gls{QSI} generally requires large amounts of data and requires very strong diffusion gradients. Henceforth, the implementation of true \gls{QSI} on clinical systems is very difficult and the derived parameters are often difficult to interpret (as seen in Chapters \ref{chapter 5}\&\ref{chapter 6}).
\paragraph{}
In the following chapters we will investigate a different model-based diffusion imaging technique, which can be seen as a compromise between the simple model of \gls{DTI} and the complex \gls{QSI} approach. Here, our work is heavily based on the ActiveAx approach by \citet{Alexander:2008}, who used a simple compartment tissue model to derive axon diameter and axon density indices in the brain using standard clinical hardware (see Chapter\ref{ref}).


Alexander's work is designed under the premise of full brain imaging and therefore he presents DWI protocols that work independently of the fibre orientation. We call this the orientation invariant ({\OI}) approach. The {\OI} approach is based on acquiring multiple HARDI shells and requires good angular resolution to retain the orientation invariant features. As a consequence, the {\OI} protocols used e.g. in \citep{Alexander:2010} require scan times of more than 1h, which limits their transition to clinical practice.


The main motivation of this work is to reduce the scan time of the ActiveAx protocols to a more clinically applicable limit of 20-30 minutes. The central objective in attaining this reduction is to discard the requirement for orientational invariance and focus on structures with a coherent known single fibre orientation. Many previous studies have focussed on those single fibre structures (e.g. \gls{CC} or \gls{SC}) as they can be expressed in terms of relatively simple models, without the need to include complex fibre configurations such as multiple fibre crossings, bending or fanning. Furthermore, many single fibre structures such as \gls{CC} and \gls{SC} are of great importance to many developmental and pathological processes. Our approach makes imaging these structures more practicable within a 20-30 minute time frame, where a >1h. full-brain protocol might not be feasible.
\paragraph{}
In this chapter, we present a modification to the existing protocol optimization framework of \cite{Alexander:2008} (see Section~\ref{sec:protocol_optimisation}) that incorporates this a-priori information about fibre organisation. We refer to this approach and its resulting protocols as the single-fibre ({\SF}) method. We evaluate different {\SF} implementations and use computer simulations to test and compare the \SF method to the {\OI} approach. By design, the {\SF} is optimised only for a specific predetermined fibre orientation. However, in reality there will always be some variation even in very coherently aligned \gls{WM} tracts. We therefore test how much influence this variation has on our {\SF} approach compared to {\OI}. Finally, we implemented the {\SF} protocol to demonstrate its feasibility within a real biological tissue sample of fixed cervical monkey \gls{SC}. Although this study is motivated by improving \emph{in-vivo} clinical applicability, we chose to first test our approach on a pre-clinical scanner on a fixed tissue sample to avoid the challenges of \emph{in-vivo} scanning (e.g. motion and insufficient image resolution) and focus on an initial demonstration of our method. The results and observations made in this chapter form the basis of the subsequent \emph{in-vivo} implementation we present in the following chapter.

\section{Protocol optimisation}
\label{sec:chap7 protocol optimisation}
This section discusses how we adapt the {\OI} optimisation of \citet{Alexander:2008} to the {\SF} case. The original {\OI} optimisation is designed to determine the best PGSE pulse parameters (i.e. \gls{gstr}, \gls{smalldel}, \gls{bigdel}) for a fixed number $M$ of shells, each including a fixed number $N$ of distributed gradient directions. The gradient directions scheme in the \OI{} approach is fixed and uniformly distributed over the sphere as in \citep{Cook:2007}. The HARDI-like gradient scheme is necessary to guarantee orientational invariance, i.e. be sensitive to restriction regardless of diffusion direction. In our {\SF} implementation we can assume a known diffusion direction and therefore do not require the uniform angular sampling of the {\OI} approach. This allows us to include the gradient scheme together with the PGSE pulse setting in the optimisation process. In addition to {\SF}, which optimises gradient scheme and PGSE pulse parameters simultaneously, we are also interested in how much effect the optimisation of the gradient scheme ({\DO}) alone and the PGSE parameters alone ({\SD}) have on the performance of the resulting protocols. In total we tested the four following different variants of \citet{Alexander:2008}'s optimisation framework:

\begin{description}
	\item[\OI:] The orientationally invariant protocol optimisation as described in \citep{Alexander:2008}.
	\item[\FD:] The adapted algorithm optimises both the PGSE parameters and gradient direction scheme for $M$ shells assuming a single a-priori known fibre direction.
	\item[\SD:] Similar to the {\OI} optimisation approach, the algorithm uses a fixed gradient scheme for each of the $M$ shells and only optimises the PGSE pulse parameters. The main difference to \OI{} is that the algorithm optimises for a single a-priori known fibre direction instead of a set of \lq worst case\rq fibre directions.
	\item[\DO:] This is a two-step optimisation approach. The {\SD} algorithm is performed to determine the best $M$ PGSE pulse settings. In a second step the gradient scheme for each of the each $M$ shells is optimised, keeping the PGSE pulse parameters fixed.
\end{description}

Table~\ref{tab:chapter7 exp1 opt parameter overview} summarises the optimisation parameters for each of the four algorithms. It is important to note that the inclusion of the optimisation of individual gradient directions in {\DO} and {\FD} drastically increases the number of free parameters to be optimised from $O(M)$ to $O(N\cdot M)$ compared to {\OI} and {\SD}.
\begin{table}[th]
  	\caption{Overview of parameters for the tested protocol optimisation approaches.} \centering \vspace{2ex}
    \label{tab:chapter7 exp1 opt parameter overview}
    \begin{adjustbox}{width={\textwidth},totalheight=\textheight,keepaspectratio}
        	\begin{tabular}
        		{@{}y{0.35\textwidth}x{0.2\textwidth}x{0.2\textwidth}x{0.2\textwidth}x{0.2\textwidth}x{0.15\textwidth}@{}}
                \toprule
                Protocol name & {\OI} & {\FD} & {\SD} & {\DO}  \\%
        		& \footnotesize{(orientation invariant)} & \footnotesize{(pulses + directions)} & \footnotesize{(pulses only)} & \footnotesize{(directions only)} \\%
        		\cmidrule(r){2-2} \cmidrule(rl){3-3} \cmidrule(rl){4-4} \cmidrule(l){5-5} Free parameters & $\delta_m,\Delta_m,G_m$ & $\phi_{m,n},\theta_{m,n}$ &  $\delta_m,\Delta_m,G_m$ & $\phi_{m,n},\theta_{m,n}$ \\
        		& & $\delta_m,\Delta_m,G_m$ & &  \\[1.5ex]
        		Fixed parameters & $\phi_n,\theta_n$ & -- & $\phi_n,\theta_n$ & $\delta_m,\Delta_m,G_m$  \\[2ex]
        		Specific fibre direction & no & yes & yes & yes \\[2ex]
        		No. of free parameters & O($M$) & O($M\times N)$ &O($M$) & O($M\times N$) \\
        		\bottomrule
        	\end{tabular}        	
    \end{adjustbox}
\end{table}

\subsection{Tissue model}
   Like \citep{Alexander:2008}, we use a simple two-compartment model of white matter tissue, with axons represented as cylinders with a single radius and extra-axonal tissue represented by an axially symmetric diffusion tensor ('ZeppelinCylinder' in the taxonomy of \cite{Panagiotaki:2012}).  For the optimisation we assume tissue diffusion properties typically found in \emph{in-vivo} human white matter. A full list of the model parameters used in the optimisation is given in Table~\ref{tab: chapter7 exp1 model opt parameters clinical}.

\begin{table}[!ht]\scriptsize
	\caption{Model parameters used for optimisation.}
    \centering
    \vspace{2ex}	
    % Cylinder
    \begin{tabular}{y{0.1\textwidth}y{0.25\textwidth}}
    \addlinespace
    \toprule
    \multicolumn{2}{c}{\textbf{Cylinder}} \\
    \multicolumn{2}{c}{(intra-axonal)} \\
    \midrule
    $f_{intra}$ & $0.7$ \\
    $d_{\|}$ & $1.7\cdot 10^{-9}$ $s/mm^2$ \\
    $d_{\bot} $ & $6\cdot 10^{-10}$ $s/mm^2$ \\
    dir   & $[0,0,1]^T$ \\
    R & $\{1..1..10\}\mu m$ \\
    \bottomrule
    \end{tabular}%
    \hspace{0.15\textwidth}
    %%% Zeppelin
    \begin{tabular}{y{0.1\textwidth}y{0.25\textwidth}}
    \addlinespace
    \toprule
    \multicolumn{2}{c}{\textbf{Axially symmetric tensor}} \\
    \multicolumn{2}{c}{(extra-axonal)} \\
    \midrule
    $f_{extra}$& $0.3$ \\
    $d_{\parallel}$ & $1.7\cdot 10^{-9}$ $m^2/s$ \\
    $d_{\bot} $ & $6\cdot 10^{-10}$ $s/m^2/s$ \\
    dir & $[0,0,1]^T$ \\
    \bottomrule
    & \\ %empty line to level tables
    \end{tabular}%
    \label{tab: chapter7 exp1 model opt parameters clinical}
    \vspace{2ex}	
\end{table}

\subsection{Implementation}
We implement the four algorithms as described in Section~\ref{sec:chap7 protocol optimisation} for $M=8$ different sets of PGSE parameters. For for {\OI} and {\SD} we use the default setting for the SOMA algorithm , i.e. 500 migration steps with 100 migrating individuals. For {\DO} and {\SD}, the parameter space to explore is much larger and we are using 1000 individuals with 750 migration steps to ensure convergence. To avoid local minima, each optimisation is repeated 5 times and from the resulting protocols the one with the smallest CRLB is chosen. The available gradient strength is limited to 60mT/m to simulate a typical clinical MRI scanner.

\subsection{Optimised protocols}
Figure~\ref{fig:chapter7 exp1 DIRS vs CRLB} shows the CRLB of the four different optimisation schemes for different $N$. We can see that CRLB distinguishes the four protocols in two groups: {\OI}/{\SD} with fixed directions and {\DO}/{\FD} with optimised directions. {\DO} and {\FD} consistently produce lower CRLB than {\OI} and {\SD} regardless of the chosen $N$, although the differences between the two groups become smaller for larger $N > 60$. For better comparison, Figure~\ref{fig:chapter7 exp1 DIRS vs CRLB} shows the corresponding $N$ for a different values of CRLB. We can see that {\FD} and {\DO} both achieve the same CRLB values of {\OI} with significantly less directions. In other words, the {\DO} and {\FD} promise a comparable performance to {\OI} with only 25\%--33\% the amount of data required.

\begin{figure}[H]
	\centering
	\input{chapter7/figs/SNRvsCRLB_60mT.tex}
	%\input{chapter7/figs/SNRvsCRLB_300mTCopenhagen.tex}
	\caption{CRLBs of optimised protocols under different SNR.}
	\label{fig:chapter7 exp1 SNR vs CRLB}
\end{figure}

Figure~\ref{fig:chapter7 exp1 SNR vs CRLB} shows the CRLB of the four different optimisations for a fixed $N=90$ but different \gls{SNR} conditions. The results here are very similar to the findings shown above. Both {\FD} and {\DO} consistently show CRLB values three to four times lower than those of {\OI} and {\SD} for similar \gls{SNR} values. The differences are most apparent for low \gls{SNR} values, which are also more common in real clinical diffusion MRI.
\begin{figure}[H]
	\centering
	\input{chapter7/figs/DIRSvsCRLB_60mT.tex}
	%\input{chapter7/figs/DIRSvsCRLB_300mTCopenhagen.tex}
	\caption{CRLBs of optimised protocols with different $N$  number of gradient directions.}
	\label{fig:chapter7 exp1 DIRS vs CRLB}
\end{figure}
\paragraph{}
Table~\ref{fig:chapter7 exp1 protocol 60mT} shows an example of the \OI{} and the three \SF{} protocol variants we obtain for $N=90$ and SNR$=20$. A more comprehensive visualisation of the different protocol parameters can be found in Appendix~\ref{app:chapter7 protocols 60mT}. We deliberately chose a large number of $M$ to reevaluate the optimal number of different PGSE for each protocol. Our findings here confirm the results of \citet{Alexander:2008}, which suggest that a smaller number of $M=4$ suffices to capture all unique combinations of PGSE pulse parameters. We also note that {\OI} and {\SD} protocols are almost identical, which suggest that the choice of diffusion orientation for the {\OI} has little influence on the outcome of the algorithm.
\begin{table}[H]
\caption{Protocol parameters and gradient schemes for $N=90$, $M=8$. The $M$ different gradient direction schemes are coded by different colors. Please note that in the \OI{} and \SD{} protocols all $M$ gradient direction schemes are identical.}
\label{fig:chapter7 exp1 protocol 60mT}
\begin{adjustbox}{width=0.8\linewidth,keepaspectratio,minipage=\linewidth}
\tiny
\subfloat[\OI protocol]
{
    \begin{minipage}{\textwidth}
    \begin{tabular}{rrrr}
        \addlinespace
        \toprule
        $\delta$ & $\Delta$ & $|G|$ & $b$-value \\
        $[ms]$ & $[ms]$ & $[mT/m]$ & $[s/mm^2]$\\
        \midrule
        12.9  & 17.9  & 60    & 583 \\
        12.9  & 17.9  & 60    & 585 \\
        12.9  & 17.9  & 60    & 587 \\
        12.9  & 17.9  & 60    & 588 \\
        22.7  & 48.3  & 30    & 1480 \\
        24.4  & 46.6  & 30    & 1684 \\
        26.3  & 44.7  & 40    & 2752 \\
        33    & 38    & 60    & 7585 \\
        \bottomrule
    \end{tabular}%
    \hspace{1cm}
        \raisebox{-2cm}{
        \begin{tikzpicture}[scale=0.7]
              \begin{axis}[xmax=1, xmin=-1, ymax=1, ymin=-1, zmax=1, zmin=-1, width=4.0cm, height=4.0cm, grid=major,legend entries={all},cycle multi list={color list\nextlist[2 of]mark list},legend pos=outer north east]
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_OI_60mT001.txt};
        \end{axis}
        \end{tikzpicture}
        }
  \end{minipage}
}



\subfloat[\FD protocol]
{
    \begin{minipage}{\textwidth}
    \begin{tabular}{rrrr}
    \addlinespace
    \toprule
        $\delta$ & $\Delta$ & $|G|$ & $b$-value \\
        $[ms]$ & $[ms]$ & $[mT/m]$ & $[s/mm^2]$\\
        \midrule
        14.7  & 19.8  & 60    & 832 \\
        14.8  & 19.8  & 60    & 837 \\
        15.1  & 20.1  & 60    & 883 \\
        15.4  & 20.4  & 60    & 932 \\
        21.2  & 53.9  & 30    & 1361 \\
        27.2  & 47.9  & 40    & 3123 \\
        27.9  & 47.2  & 40    & 3452 \\
        35    & 40    & 60    & 8974 \\
        \bottomrule
    \end{tabular}%
    \hspace{1cm}
        \raisebox{-2cm}{
        \begin{tikzpicture}[scale=0.7]
              \begin{axis}[xmax=1, xmin=-1, ymax=1, ymin=-1, zmax=1, zmin=0, width=4.0cm, height=4.0cm, grid=major,legend entries={1,2,3,4,5,6,7,8},cycle multi list={color list\nextlist[2 of]mark list},legend pos=outer north east]
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_FD_60mT005.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_FD_60mT004.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_FD_60mT003.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_FD_60mT001.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_FD_60mT007.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_FD_60mT008.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_FD_60mT006.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_FD_60mT002.txt};
        \end{axis}
        \end{tikzpicture}
        }
    \end{minipage}
}


\subfloat[\SD protocol]
{
  \begin{minipage}{\textwidth}
    \begin{tabular}{rrrr}
        \addlinespace
        \toprule
        $\delta$ & $\Delta$ & $|G|$ & $b$-value \\
        $[ms]$ & $[ms]$ & $[mT/m]$ & $[s/mm^2]$\\
        \midrule
        12.8  & 17.8  & 60    & 572 \\
        12.9  & 17.9  & 60    & 580 \\
        12.9  & 17.9  & 60    & 580 \\
        12.9  & 17.9  & 60    & 582 \\
        21.9  & 48.9  & 30    & 1484 \\
        24.7  & 46    & 30    & 1677 \\
        26.3  & 44.5  & 40    & 2639 \\
        32.9  & 37.9  & 60    & 7515 \\
        \bottomrule
    \end{tabular}%
    \hspace{1cm}
        \raisebox{-2cm}{
        \begin{tikzpicture}[scale=0.7]
              \begin{axis}[xmax=1, xmin=-1, ymax=1, ymin=-1, zmax=1, zmin=-1, width=4.0cm, height=4.0cm, grid=major,legend entries={all},cycle multi list={color list\nextlist[2 of]mark list},legend pos=outer north east]
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_OI_60mT001.txt};
        \end{axis}
        \end{tikzpicture}
        }
  \end{minipage}
}


\subfloat[\DO protocol]
{
    \begin{minipage}{\textwidth}
        \begin{tabular}{rrrr}
            \addlinespace
            \toprule
            $\delta$ & $\Delta$ & $|G|$ & $b$-value \\
            $[ms]$ & $[ms]$ & $[mT/m]$ & $[s/mm^2]$\\
            \midrule
            12.8  & 17.8  & 60    & 572 \\
            12.9  & 17.9  & 60    & 580 \\
            12.9  & 17.9  & 60    & 580 \\
            12.9  & 17.9  & 60    & 582 \\
            21.9  & 48.9  & 30    & 1484 \\
            24.7  & 46    & 30    & 1677 \\
            26.3  & 44.5  & 40    & 2639 \\
            32.9  & 37.9  & 60    & 7515 \\
            \bottomrule
        \end{tabular}%
        \hspace{1cm}
        \raisebox{-2cm}{
        \begin{tikzpicture}[scale=0.7]
              \begin{axis}[xmax=1, xmin=-1, ymax=1, ymin=-1, zmax=1, zmin=0, width=4.0cm, height=4.0cm, grid=major,legend entries={1,2,3,4,5,6,7,8},cycle multi list={color list\nextlist[2 of]mark list},legend pos=outer north east]
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_DO_60mT004.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_DO_60mT007.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_DO_60mT003.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_DO_60mT008.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_DO_60mT006.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_DO_60mT001.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_DO_60mT002.txt};
              \addplot3+[only marks] table {chapter7/figs/protocols_par1/SF_DO_60mT005.txt};
        \end{axis}
        \end{tikzpicture}
        }
    \end{minipage}
}
\end{adjustbox}
\end{table}

\section{Experiments}
\subsection{Synthetic data}
From all protocols for $N=90$ and $SNR=20$ we generate synthetic datasets, with which we then test and compare the performance of the different protocol optimisations. The number of gradient directions was chosen to be large in order to allow sufficient angular resolution for {\OI} and {\DO} protocols.
\paragraph{}
\label{sec: chapter 6 fibre distribution perfect}
The MR signals are generated from the same tissue model and tissue diffusion properties we used for the protocol optimisation (see section \ref{sec:chap7 protocol optimisation} and Table~\ref{tab: chapter7 exp1 model opt parameters clinical} for details). We use the Camino software package \citep{Cook:2006} to simulate datasets for four different cylinder radii $R={1,2,5,10}\mu m$ as a representation of radii usually found in \emph{in-vivo} white matter tissue. To simulate noise in the MR acquisition, we add Rician noise with $SNR=20$ to the simulated noise-free datasets. To concentrate on comparing the estimates of radius and volume fraction between the protocols we assume all cylinders to be perfectly aligned along the $[0,0,1]^T$ as assumed in the optimisation process.

\subsubsection*{Effect of principle diffusion direction}
Even in very coherent structures such as the \gls{CC}, the observed fibre direction is subject to some variation. However the protocols {\FD} and {\DO} contain gradient schemes designed for a fixed principle diffusion direction. To test the effect of potential misalignment between assumed and true principal diffusion direction we designed the following experiment.
\paragraph{}
We created similar dataset as above, with the exception of varying the principal diffusion direction for each synthetic sample. Using the spherical representation of the cylinder direction, we sampled each combination of polar angle $\phi \in \{0..90\}^\circ$ and azimuth angle $\theta \in \{0..180\}^\circ$ in discrete steps of $5^\circ$. In total we simulated 703 unique datasets with different principal diffusion directions (see Figure~\ref{fig:chapter7 exp1 Ang Error dirs} for a plot of the generated diffusion directions).
\bgroup
\tikzset{use png}

\begin{figure}[h]
    \centering
	  \ifdraft
	  {
	  	\draftpic
	  }
	  {
	  	\input{chapter7/figs/AngErrorDirSamples.tex}
	  }
	
        \caption{Generated diffusion directions to test orientation bias as 3-d scatter plot (left panel) and polar scatter plot (right panel).}
    \label{fig:chapter7 exp1 Ang Error dirs}
\end{figure}
\egroup

\subsubsection*{Dispersion}
The dispersion of fibres is another confounding factor that has considerable influence on the estimates of cylinder radius and volume fraction \cite{Zhang:2011}. While \citeauthor{Zhang:2011} also showed that the effect in coherent structures with moderate dispersion, such as in the CC, is neglible for {\OI}, it is unclear whether this is still the case for the optimised gradient schemes used in our {\SF} and {\DO} protocols. We therefore created a set of data to test the effect of dispersion on our protocols. To simulate different degrees of dispersed diffusion directions, we used the the Van-Mises-Fisher distribution $p_{d}$:
\begin{equation}
	p_{d}(\mathbf{x}; \mu, \kappa)=\frac {\kappa} {2\pi(e^{\kappa}-e^{-\kappa})} \left( {\kappa \mu^T \mathbf{x} } \right),
\end{equation}
which is defined for each point $\mathbf{x} \in \mathbb{R}^2$ on the unit sphere with $\mu$ being the mean distribution of the distribution, and $\kappa$ being the scalar concentration parameter that controls the dispersion of the distribution. From $p_d$ we sampled 1000 directions. We then took the average of the noise-free MR signal generated from our tissue model, and added Rician noise as described above. To simulate different degrees of dispersion we varied $\kappa \in [2,4,8,\dots,512]$ with fixed $\mu=[0,0,1]^T$. Figure~\ref{fig:chapter7 exp1 VanMises draw} illustrates the sampled directions for each of the used $\kappa$ values. For $\kappa\rightarrow\infty$ the amount of dispersion becomes neglible and the simulation is equivalent to the perfectly coherent fibre configuration.

\bgroup
\tikzset{use png}
\begin{figure}[h]
    \centering 	
	\input{chapter7/figs/DispersionPolar1000sam.tex} \caption{Visualisation of 1000 sample directions drawn from Von-Mises-Fisher distribution with varying $\kappa$. Left half of each figure shows the 3-d scatter plot, right half shows a polar plot of the sampled directions for each value of $\kappa$.}
  \label{fig:chapter7 exp1 VanMises draw}
 \end{figure}
\egroup
\subsection{Feasibility study on fixed monkey spinal cord}
Scans were performed on an experimental 4.7 Tesla scanner (Varian Inc., Palo Alto, CA, USA) in collaboration with the Danish Research Centre for Magnetic Resonance (DRCMR). To make best use of the hardware, we carried out the protocol optimisation as described above but increased $|G|_{max}$ to $300mT/m$. We also adjusted for differences in tissue properties between live and fixed tissue, e.g. decreased diffusivity and shrinkage of axon diameters, based on preliminary \gls{DTI} analysis of the tissue sample. The full list of modified tissue parameters is given in Table~\ref{tab: chapter7 exp1 model opt parameters ex-vivo}. Scan time in this experiment was limited to 12h only. To retain acceptable image resolution and \gls{SNR} within the given time frame, we reduced the number of acquisitions to $M=4$ and $N=30$. Furthermore, we only had time to test the {\FD} protocol in this experiment. The resulting protocol is summarised in Table~\ref{TODO} and described more comprehensively in Appendix \ref{app:chapter7 protocols 300mT}. 
\paragraph{}
We used a volume coil for transmission and a homemade 20 mm surface coil was used as receive coil. The data was was acquired using a spin-echo sequence with single line readout and a conventional pulsed gradient spin-echo preparation \citep{Stejskal:1965}.  TE/TR were 59/2000 ms, \gls{FOV} was $10\times 10 mm^2$. The matrix size was 64$\times$64 and was 2-D interpolated to 128$\times$128 leading to an axial in-plane resolution of 79 $\times$ 79 $\mu$m$^2$. To support the high in-plane resolution we acquired thick slices of 1.5mm to achieve acceptable \gls{SNR} in our data. The protocol was repeated 4 times with a total imaging time of 13 h. The magnitude images of the four repeated measurements were averaged offline prior to analysis.

\begin{table}[!ht]
	\caption{Adjusted \emph{ex-vivo} tissue model parameters used for pre-clinical scan optimisation (changes to \emph{in-vivo} protocol are displayed in red).}
    \centering
    \vspace{2ex}	
    % Cylinder
    \begin{tabular}{y{0.1\textwidth}y{0.25\textwidth}}
    \addlinespace
    \toprule
    \multicolumn{2}{c}{\textbf{Cylinder}} \\
    \multicolumn{2}{c}{(intra-axonal)} \\
    \midrule
    $f_{intra}$ & $0.7$ \\
    $d_{\|}$ & {\color{red}$4\cdot 10^{-10}$ $s/mm^2$} \\
    $d_{\bot} $ & {\color{red}$2.5\cdot 10^{-10}$ $s/mm^2$} \\
    dir   & $[0,0,1]^T$ \\
    R & {\color{red}$\{0.5,1,2\}\mu m$} \\
    \bottomrule
    \end{tabular}%
    \hspace{0.15\textwidth}
    %%% Zeppelin
    \begin{tabular}{y{0.1\textwidth}y{0.25\textwidth}}
    \addlinespace
    \toprule
    \multicolumn{2}{c}{\textbf{Axially symmetric tensor}} \\
    \multicolumn{2}{c}{(extra-axonal)} \\
    \midrule
    $f_{extra}$& $0.3$ \\
    $d_{\parallel}$ & {\color{red}$4\cdot 10^{-10}$ $m^2/s$} \\
    $d_{\bot} $ & {\color{red}$2.5\cdot 10^{-10}$ $s/m^2/s$} \\
    dir & $[0,0,1]^T$ \\
    \bottomrule
    & \\ %empty line to level tables
    \end{tabular}%
    \label{tab: chapter7 exp1 model opt parameters ex-vivo}
    \vspace{2ex}	
\end{table}

\subsection{Model fitting}
A voxel-wise fit of the MR signal is performed using a fitting method similar to \citet{Alexander:2010}. We define the objective function as the maximum likelihood of model parameters given the observed MR signals under Rician noise ($\sigma=0.05$). We use the multi-run fitting routine of \citet{Panagiotaki:2012} to find an initial parameters, which determines the best parameter estimates from multiple gradient-descend runs with pertubed starting points (n=20 in our case). Based on the initial parameter estimates, we then performed a \gls{MCMC} algorithm to determine the posterior distributions of the model parameters. To ensure convergence, we used the following, rather conservative setting for the \gls{MCMC} algorithm: burn-in of 5000, sample interval of 1000, 5\% parameter pertubations with uniform, uninformative priors.

To increase stability of the fitting we fixed $d_\parallel$ to its true value of $1.7mm/s^2$ similar to previous studies \citep{Assaf:2008,Barazany:2009,Alexander:2010}. Moreover, $d_\bot$ is approximated by the tortuosity formulation of \citet{Szafer:1995}:
$$
    D_{\bot} = f_{intra}\cdot D_{\parallel}.
$$


For {\DO} and {\FD}, we also found our method to be very unstable with respect to determining the correct principal diffusion direction of the model, even when the fibre direction was close to $[0,0,1]^T$. This possibly stems from the fact that the initial guess of the direction is computed from the \gls{DT} model. While this method works well for {\OI} and {\SD} protocols, it fails when using the optimised gradient direction sets of {\DO} and {\FD}. We therefore fix the principal fibre direction to $[0,0,1]^T$ in all experiments, unless stated otherwise.

\subsubsection*{Fixed tissue}
We chose to scan a sample of fixed monkey cervical spinal cord to test our protocol in a real biological system. The details of the sample preparation are described in \citep{Lundell:2011}. For the fixed tissue we use a slightly modified fitting routine to account for the differences between fixed and live tissue. Here we fix $d_{\|}$ to $0.45\mu m/mm^2$. The volume fraction of the restricted compartment f was constrained to be in the range of $[0.5, 1.0]$. In each voxel, the mean of the posterior distribution of $R$ and $f_{intra}$ are calculated. For consistency with indices derived from histological examination \citep{Keyserlingk:1984,LaMantia:1990a,Aboitiz:1992} and also previous imaging studies \citep{Alexander:2010}, maps of the axon diameter index $a=2\cdot R$ and the axonal density index $\rho=f_{intra}/\pi/R^2$ are reported.

\FloatBarrier

\section{Results}
\subsection{Accuracy and precision of parameter estimates}
Figure~\ref{fig:chapter7 exp1 boxplots 60mT} shows box plots of the fitted volume fraction $f_{intra}$ and $R$ for different cylinder radii.
\begin{figure}[H]
	\centering
	\input{chapter7/figs/BW_f_60mT.tex}
	\input{chapter7/figs/BW_R_60mT.tex}
	\caption{Boxplots of estimated $f_{intra}$ and $R$ for different cylinder radii.}
	\label{fig:chapter7 exp1 boxplots 60mT}
\end{figure}


\paragraph{} All protocols can successfully recover the correct value for $f_{intra}$ for all radii 1, 2 and 5 $\mu m$ with the posterior distributions being centred around the true value $f_{intra}=0.7$ with little variance. When $R=10\mu m$ we observe significantly more variance in the estimates for in all protocols. Further we find also find the estimates to be less accurate, mainly in {\SF} and {\SD}, which overestimate $f_{intra}$ slightly.


Compared to $f_{intra}$ results, we observe more variance in the posterior distributions of $R$ estimates for all protocols. Over all $R$ {\DO} and {\FD} appear to achieve better results than {\OI} and {\SD} protocols. To illustrate this better, Figure~\ref{fig:chapter7 exp1 histograms 60mT} shows the posterior distributions of $R$ for each protocol in more detail.

\begin{figure}[H]
	 \centering
	 \input{chapter7/figs/Histograms_60mT.tex}
	 \caption{Histograms of fitted cylinder radius $R$ for different protocols.}
	 \label{fig:chapter7 exp1 histograms 60mT}
\end{figure}

All protocols perform best for $R=5\mu m$ with tight posterior distributions centred around the ground truth. While generally all protocols show very good accuracy in estimating $R$, {\DO} and {\FD} have the advantage in terms of precision as their distributions are considerably tighter than for {\OI} and {\SD}. Compared to $R=5\mu m$, $R=10\mu m$ estimates are generally broader and with a noticeable  negative skew, particularily for {\OI} and {\SD} protocols.  As for $5\mu m$, {\DO} and {\FD} produce more sharp parameter distributions than {\OI} and {\SD}, although the difference is less obvious. The broader distributions for $R=10\mu m$ might be explained by the long diffusion, which are necessary to estimate large radii accurately.  We assume a typical T2 signal decay of 70ms in this experiment. As shown in \citep{Alexander:STEAM}, in this regime long diffusion times can not be reached with acceptable \gls{SNR}. As a consequence, for $R=10\u m$ a greater proportion of the signal will appear unrestricted and causes the underestimation of $f_{intra}$ and larger variation in radius estimates in our results.

Smaller radii, $R=1\mu m$ and $R=2\mu m$, are less well distinguished from each other in all protocols, with posterior distributions showing significant overlap as expected from limited gradient strength used in this experiment \citep{Alexander:2008,Alexander:2010,Siow:2012}. For small radii the \gls{COM} effect on small radii prohibits any contrast between different $R$, which we find reflected in our posterior distributions (see Chapter~\ref{TODO} for a detailed explanation). However, {\FD} suffers least from this and clearly outperforms the other protocols as it allows much better distinction between the small radii.


\subsection{Variation of fibre direction}
Figure~\ref{fig:chapter7 exp1 angular vals 60mT} shows the root mean squared error for both $f_{intra}$ and $R$ parameters with respect to the principle orientation of diffusion for cylinders with small radius $R=2\mu m$ and large radius $R=5\mu m$. As in the previous experiment, all the protocols generally perform better for $R=5\mu m$ than $R=2\mu m$. Therefore, the most marked differences between protocols appear in the $R=2\mu m$ dataset.


The plots clearly show that estimates of $f_{intra}$ are unaffected by the true diffusion orientation in all protocols for $R=2\mu m$. While this is expected for {\OI} and {\SD}, we can explain the performance of {\FD} and {\DO} by the small cylinder radius, which allows {\FD} and {\OI} to distinguish hindered and restricted diffusion even when the gradient scheme is not aligned perpendicular to the intra-axonal compartment. Results for $R=5\mu m$ show that $f_{intra}$ estimates are affected more by the misalignment of the gradient scheme with larger radii, although the estimation error still remains low over all different directions.


Larger differences between the protocols can be seen in the $R$ estimation error. Again, {\OI} and {\SD} are consistent over the whole range of simulated fibre directions, with no apparent directional pattern of the estimation error. In contrast, {\FD} and {\DO} performance is considerably more impeded for larger misalignment of gradient scheme and cylinder orientation. The maximum estimation error is observed for rotation angles around $45^\circ$. With $\phi$ close to $90^\circ$, the {\SF} and {\DO} gradient direction scheme now features more measurements perpendicular and parallel to the cylinder compartment, which reduces again the fitting error. For smaller misalignments of $\phi<15^\circ$ both {\FD}/{\DO} perform very similar to the {\OI}/{\SD} protocols.

\begin{figure}[!h]
\centering
	\subfloat[Cylinder radius $R=2\mu m$]
    {
        \begin{minipage}{\textwidth}    
        	\pgfimage[width=0.9\textwidth]{chapter7/figs/errs_2D_f_60mT.pdf}\\
            \pgfimage[width=0.9\textwidth]{chapter7/figs/errs_2D_R_60mT.pdf}
        \end{minipage}
    }
    
    
    
    \subfloat[Cylinder radius $R=5\mu m$]
    {
        \begin{minipage}{\textwidth}    
            \pgfimage[width=0.9\textwidth]{chapter7/figs/errs_2D_f_60mTR5.pdf}\\
            \pgfimage[width=0.9\textwidth]{chapter7/figs/errs_2D_R_60mTR5.pdf}
        \end{minipage}
    }
  	\caption{Root mean square error of $f_{intra}$ and $R$ estimates for different principle diffusion directions.}
	\label{fig:chapter7 exp1 angular vals 60mT}
\end{figure}


\paragraph{Effect of fibre dispersion: } Figure~\ref{fig:chapter7 exp1 dispersion 60mT} shows the fitted $f_{intra}$ and $R$ estimates different degrees of intra-voxel dispersion of fibre orientations. For small $\kappa$, i.e. high degrees of dispersion, we observe large errors in estimates of both $f_{intra}$ and $R$, as expected e.g. from \citep{Zhang:2011}. However, for $\kappa \ge 64$, the fitted parameters converge to their respective ground truth, despite a considerable degree of dispersion still present in those datasets. With increasing $\kappa$, there appears to be no noticeable difference between the four protocols.

\begin{figure}[ht]
	\centering
	\input{chapter7/figs/BW_Dispersion_f_300mT.tex}
	\input{chapter7/figs/BW_Dispersion_R_300mT.tex}
	\caption{Mean and standard deviation of posterior distribution for fitted volume fraction and radius for different $\kappa$ values.}
	\label{fig:chapter7 exp1 dispersion 60mT}
\end{figure}

\subsection{Proof-of-concept \emph{ex-vivo} implementation}
Figure~\ref{fig:chapter7 exp1 monkey data} presents maps of axon diameter $a$ and axonal density $\rho$ in the upper cervical spinal cord obtained from the {\FD} protocol. We can discriminate clearly differences in axon diameter and axonal density indices between anatomically different white matter tracts. Dorsal and lateral sensory tracts show small axon diameters between 1--4$\mu m$ and a density of 0.03--0.08$\mu m^{-2}$. The smallest axon calibers (<1.5$\mu m$) are observed in the dorsal columns (DC) while mean axon size in the anterolateral column (ALC) is 1.5--2.5$\mu m$. The largest axons (3--4$\mu m$) are found in the corticospinal tracts (CST) together with low density of 0.01--0.02$\mu m^{-2}$. Overall, we clearly observe left-right symmetry of axon diameter and density in all tracts, which corresponds well with the known anatomical organisation of the the SC. Parameters are also consistent along the SC within the limits of anatomical variation and are in the range of values reported in previous histological evaluation of the cord \citep{GrafvonKeyserlingk:1984,Golabchi:2010}. 

\begin{figure}[ht]
	\centering
	\subfloat[Axon diameter index $a$]
	{
   	  \ifdraft
	  {
	  	\draftpic
	  }
      {
		\pgfimage[width=4cm]{chapter7/figs/monkeySC_diam.png}
      }   
	}
	\hspace{2cm}
	\subfloat[Axon density index $\rho$]
	{
      \ifdraft
	  {
	  	\draftpic
	  }
      {
		\pgfimage[width=4cm]{chapter7/figs/monkeySC_dens.png}
      }
	}
	\caption{Axial slice of upper cervical cord showing maps of $a$ and $\rho$. White markers show the approximate location of the corticospinal tracts (CST), anterolateral column (ALC) and dorsal column (DC)}
	\label{fig:chapter7 exp1 monkey data}
\end{figure}

\section{Discussion}
We presented here a novel method that provides optimal diffusion weightings and gradient directions for estimating axonal diameter by exploiting the single fibre orientation of structures like the corpus callosum or spinal cord. We demonstrated that the {\SF} approach reduces the required amount of data by 60--75\%, while achieving similar performance to {\OI}. This translates into a considerable reduction in scan time from >1h for {\OI} to less than 20-25 minutes, which makes routine clinical implementation much more viable.  

\paragraph{}
Our results suggest that a dedicated optimisation of gradient directions is clearly beneficial over the orientation invariant \OI{} approach, both in terms of CRLB and simulated noisy data when the fibre direction is known a-priori. We find that the optimisation routine deliberately diminishes angular resolution in favour of measuring diffusion only in the most informative directions (predominantly parallel and perpendicular to the WM tracts). We noticed that our optimised gradient scheme pick out predominantly gradient directions parallel and perpendicular to the cylinder compartment, presumably to maximise sensitivity to restricted diffusion. In that respect our automatically optimised gradient schemes agree with many other studies e.g. \citep{Stanisz:1997,Avram:2008,Assaf:2008,Panagiotaki:2012}, which only chose parallel and perdendicular measurements to maximise sensitivity to restricted diffusion. Our results confirm that these directions give the most information about the tissue, and one should focus on those directions when the fibre configuration is known a-priori. 


We furthermore observe that our {\FD} protocols contain slightly stronger diffusion weighting factors compared to the other protocols. We can attribute this to the custom gradient scheme, which translates to more measurements in the perpendicular direction and hence profits more from stronger diffusion attenuation than {\OI}. The simulation results show that the combination of stronger diffusion weighting and custom gradient scheme is superior in estimating the cylinder radius, especially when $R$ is very small.


Considering the small number of distinct gradient directions, it comes as no surprise that we found the optimised gradient schemes in {\FD} and {\DO} more prone to error when the true fibre orientation differs from the one assumed in the optimisation process. However, when the misalignment was less than 15$^\circ$ (as expected in coherent WM structures) the impact was neglible. We are confident that with careful positioning, the {\SF} gradient scheme can be aligned to the dominant fibre directions in CC and SC with little error without diminishing the performance of the {\SF} protocols.


Finally, the feasibility study on our post-mortem SC sample demonstrates that we can estimate a reasonable range of axon diameters and densities under realistic imaging conditions, while retaining a high image resolution. Furthermore we were able to distinguish different WM tracts of the SC by both axon diameter and axon density.
\paragraph{}
There are some limiting factors to our study. Firstly, we only optimise and fit a very simple tissue model, which only allows a crude approximation of the real tissue properties. Other, more complex, models might be a better representation of the real tissue microstructure, e.g. modelling explicitly a distribution of axon radii \citep{Barazany:2009} or including more extra-axonal tissue compartments \citep{Stanisz:1997,Wang:2011,Panagiotaki:2012}.  While these models might arguably be more accurate, they also require many more acquisitions and therefore do not agree with the aim of this study to reduce the scan time to less than 30 minutes. The choice of a simpler model offers a compromise of manageable data requirements and informative model parameters.
\paragraph{}
Secondly, the spinal cord sample we used in this study was not available for histological processing. Despite the lack of independent validation, the differences between WM tracts agree with previous results in \citep{Golabchi:2010}. It should be noted that our results were obtained with a significantly smaller number of diffusion weighted directions and b-factors compared to other studies \citep{Assaf:2008,Barazany:2009,Panagiotaki:2012}, making it more comparable to conditions found in \emph{in-vivo} scanning. Nevertheless, in many cases the pre-clinical set-up has advantages over the \emph{in-vivo} situation, as it profits from long scan times, high \gls{SNR} and an absence of motion artefacts. Although the findings in this experiment might therefore not be directly transferable to clinical practice, the results are sufficiently encouraging to pursue the \SF{} approach in clinical practice. In the next two chapters we will therefore focus on moving the \SF{} approach to towards a feasible \emph{in-vivo} implementation on standard clinical hardware.


\FloatBarrier
\begin{subappendices}

\section{Protocols}
\ifdraft
{
   Appendix here
}
{
    \subsection{Protocols optimised for $|G|_{max}$=60mT/m, N=90, M=8}
    \label{app:chapter7 protocols 60mT}
    \begin{figure}
    	[H] \centering \input{chapter7/figs/protocols_par1/OInv_62mT.tex} \caption{\OI protocol optimised for clinical gradient strength.}
       \label{fig:chapter7 exp1 OINV_62mT}
    \end{figure}
    \begin{figure}
    	[H] \centering \input{chapter7/figs/protocols_par1/SameDirs_62mT.tex} \caption{\SD protocol optimised for clinical gradient strength}
       \label{fig:chapter7 exp1 SAMEDIRS_62mT}
    \end{figure}
    \begin{figure}
    	[H] \centering \input{chapter7/figs/protocols_par1/DO_62mT.tex} \caption{\DO protocol optimised for clinical gradient strength}
       \label{fig:chapter7 exp1 DIRSONLY_62mT}
    \end{figure}
    \begin{figure}
    	[H] \centering \input{chapter7/figs/protocols_par1/FD_62mT.tex} \caption{\FD protocol optimised for clinical gradient strength}
       \label{fig:chapter7 exp1 FREEDIRS_62mT}
    \end{figure}

    \subsection{Pre-clinical scanner (300mT/m), N=30, M=4}
    \label{app:chapter7 protocols 300mT}	
    %\begin{figure}
    %	[H] \centering \input{chapter7/figs/protocols_par1/OInvariant_300mTCopenhagen.tex} \caption{OI protocol optimised for pre-clinical animal scanner} \label{fig:chapter7 exp1 INVARIANT_300mT}
    %\end{figure}
    %\begin{figure}
    %	[H] \centering \input{chapter7/figs/protocols_par1/SameDirs_300mTCopenhagen.tex} \caption{OI protocol optimised for pre-clinical animal scanner} \label{fig:chapter7 exp1 SAMEDIRS_300mT}
    %\end{figure}
    %\begin{figure}
    %	[H] \centering \input{chapter7/figs/protocols_par1/DirsOnly_300mTCopenhagen.tex} \caption{OI protocol optimised for pre-clinical animal scanner} \label{fig:chapter7 exp1 DIRSONLY_300mT}
    %\end{figure}
    \begin{figure}
    	[H] \centering \input{chapter7/figs/protocols_par1/FreeDirs_300mTCopenhagen.tex} \caption{\SF protocol optimised for pre-clinical scanner and fixed tissue.} \label{fig:chapter7 exp1 FREEDIRS_300mT}
    \end{figure}
}
%\tikzset{external/export=true}
\end{subappendices}
